### PR TITLE
bluetooth: controller: Fix command complete

### PR DESCRIPTION
**Pull request by Tschet1**
When HCI commands for legacy and extended commands are being used, a Command Disallowed error should be produced. In this case, the command complete event was missing the other (command specific) parameters in the event.

Added them.

### CODE REVIEW

File: subsys/bluetooth/controller/hci_internal.c [hunk 1]
```diff
@@ -1319,21 +1438,21 @@ int hci_internal_cmd_put(uint8_t *cmd_in)
...
+		 * up in this case.
 		 */
 		if (command_generates_command_complete_event(opcode)) {
-			uint8_t param_length = sizeof(struct bt_hci_evt_cmd_complete)
-					     + sizeof(struct bt_hci_evt_cc_status);
-			(void)encode_command_complete_header(cmd_complete_or_status.raw_event,
-							     opcode,
-							     param_length,
-							     BT_HCI_ERR_CMD_DISALLOWED);
+			generate_status_disallowed_command_complete_event(cmd_in, opcode);
```
Comment (minkrg) [inline subsys/bluetooth/controller/hci_internal.c:1450]: I might be missing something, but the opcodes in `command_generates_command_complete_event ` are less than the ones in the newly added function `generate_status_disallowed_command_complete_event `
Shouldn't they be the same?

**Comment by de-nordic** [comment]
@rugeGerritsen @minkrg Please review.

**Review by rugeGerritsen — CHANGES_REQUESTED** [review]
Discussed offline. Will see if we can combine with something in `is_host_using_legacy_and_extended_commands()` to reduce code size

File: subsys/bluetooth/controller/hci_internal.c [hunk 2]
```diff
@@ -55,51 +61,100 @@ static bool command_generates_command_complete_event(uint16_t hci_opcode)
...
 #else
 	/* A host is not allowed to use both legacy and extended HCI commands.
 	 * See Core v5.1, Vol2, Part E, 3.1.1 Legacy and extended advertising
 	 */
-	static enum {
-		ADV_COMMAND_TYPE_NONE,
-		ADV_COMMAND_TYPE_LEGACY,
-		ADV_COMMAND_TYPE_EXTENDED,
-	} type_of_adv_cmd_used_since_reset;
+	static enum type_of_adv_cmd type_of_adv_cmd_used_since_reset;
```
Comment (rugeGerritsen) [inline subsys/bluetooth/controller/hci_internal.c:114]: You could consider explicitly initializing `type_of_adv_cmd_used_since_reset` to `ADV_COMMAND_TYPE_NONE`

**Review by rugeGerritsen — APPROVED** [review]

**Review by minkrg — APPROVED** [review]

**Pull request merged**